proxy.go

```diff
@@ -80,8 +80,21 @@ func NewProxyFromConfig(logger *logrus.Logger, conf ProxyConfig) (p Proxy, err e
 	})
 
 	p.HTTPAddr = conf.HTTPAddress
-	// TODO Timeout?
-	p.HTTPClient = &http.Client{}
+
+	transport := &http.Transport{
+		MaxIdleConns:        conf.MaxIdleConns,
+		MaxIdleConnsPerHost: conf.MaxIdleConnsPerHost,
+	}
+	if conf.IdleConnectionTimeout != "" {
```

Comment:
I think this would be better as:

``` go
        var idleTimeout time.Duration
	if conf.IdleConnectionTimeout != "" {
		idleTimeout, err = time.ParseDuration(conf.IdleConnectionTimeout)
		if err != nil {
			return p, err
		}
        }
	transport := &http.Transport{
		MaxIdleConns:        conf.MaxIdleConns,
		MaxIdleConnsPerHost: conf.MaxIdleConnsPerHost,
                IdleConnTimeout = idleTimeout,
	}
```

lets you stay on the happy path (: